### PR TITLE
[MDS-5779] Added support for permit keyword search in Core

### PR DESCRIPTION
**Pull request by simensma-fresh**
## Objective 

[MDS-5779](https://bcmines.atlassian.net/browse/MDS-5779)

**Permit Search**
Added the option to search indexed permit documents from the new permit search service using the global search bar in core. Feature flagged behind the `permit_document_keyword_search` flag

image

**Changes**
- Updated the `docker-compose` command to `docker compose` as `docker-compose` is deprecated
- Typescriptified + the search results component + converted it to a functional component
- Modified Core API to also perform a permit search
- Modified the Permit Service to perform a keyword search and highlight matches
- Added in a utility script to bulk index permit documents based on a `csv` exported from Core

### CODE REVIEW

**Review by henryoforeh-dev — DISMISSED** [review]

File: services/core-web/src/tests/components/search/__snapshots__/SearchResults.spec.js.snap [hunk 1]
```diff
@@ -1,3 +1,62 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
-exports[`Search renders properly 1`] = `<Fragment />`;
+exports[`Search renders properly 1`] = `
+Object {
+  "asFragment": [Function],
+  "baseElement": <body>
+    <div />
+  </body>,
+  "container": <div />,
```
Comment (taraepp) [inline services/core-web/src/tests/components/search/__snapshots__/SearchResults.spec.js.snap]: just comparing this to other test results, would looking at container.firstChild be a bit better? Still an improvement over <Fragment />. 🤷‍♀️

**Review by asinn134 — DISMISSED** [review]
Really interesting stuff, nice work!

File: services/core-web/src/tests/components/search/SearchResults.spec.js [hunk 2]
```diff
@@ -31,7 +32,11 @@ beforeEach(() => {
 
 describe("Search", () => {
   it("renders properly", () => {
-    const component = shallow(<SearchResults {...dispatchProps} {...reducerProps} />);
+    const component = render(
```
Comment (matbusby-fw) [inline services/core-web/src/tests/components/search/SearchResults.spec.js:36]:
```suggestion
    const { container } = render(
```

File: services/core-web/src/tests/components/search/SearchResults.spec.js [hunk 3]
```diff
@@ -31,7 +32,11 @@ beforeEach(() => {
...
+      <ReduxWrapper>
+        <SearchResults {...dispatchProps} {...reducerProps} />
+      </ReduxWrapper>
+    );
     expect(component).toMatchSnapshot();
```
Comment (matbusby-fw) [inline services/core-web/src/tests/components/search/SearchResults.spec.js:41]:
```suggestion
    expect(container.firstChild).toMatchSnapshot();
```

**Review by matbusby-fw — DISMISSED** [review]
🥇

**Review by taraepp — COMMENTED** [review]
🎨

**Review by matbusby-fw — APPROVED** [review]
🚛

**Pull request merged**